config/configgrpc/configgrpc.go

```diff
@@ -167,8 +167,31 @@ type GRPCServerSettings struct {
 	Auth *configauth.Authentication `mapstructure:"auth,omitempty"`
 }
 
+type clientOptions struct {
+	ext map[config.ComponentID]component.Extension
+}
+
+// ToClientSettingOption is an option to add/change additional configuration on top of GRPCClientSettings
+// to construct GRPCClientSettings.ToClient().
+type ToClientSettingOption func(options *clientOptions)
+
+// WithExtensionsConfiguration  is passed to GRPCClientSettings.ToClient() in order to
+// extend GRPCClientSettings with the configuration provided through extensions
+// e.g Authenticator configuration via extensions
+func WithExtensionsConfiguration(ext map[config.ComponentID]component.Extension) ToClientSettingOption {
+	return func(options *clientOptions) {
+		options.ext = ext
+	}
+}
+
 // ToDialOptions maps configgrpc.GRPCClientSettings to a slice of dial options for gRPC
-func (gcs *GRPCClientSettings) ToDialOptions() ([]grpc.DialOption, error) {
+func (gcs *GRPCClientSettings) ToDialOptions(copts ...ToClientSettingOption) ([]grpc.DialOption, error) {
```

Comment:
Unclear why we use the Options idiom here. I think Extensions map is not an optional parameter. We need to make it required and pass as a regular parameter.

Reply:
Thanks @tigrannajaryan, I was concerned about a lot exporter implementations in contrib that are already using    ToDialOptions. This pattern sort of provides a backward compatibility.  Let me know,   either way I am fine.

Reply:
We are still in Beta, it is OK to break it. We just need to fix the contrib quickly and the fix is trivial, so we can do it.

Reply:
Excellent! Will make the changes.

Reply:
@tigrannajaryan, so zipkin, jaeger, promremotewrite, opencensus exporters currently use grpc client - in order to make them actually use the auth functionality they need to be refactored so that any `ToDialOptions()` calls these exporters are making  in their constructors need to be moved around into start of the exporters. 

Do you recommend we make changes in this PR? or `ToDialOptions(nil)` or `ToDialOptions(map[config.ComponentID]component.Extension{})` for now and refactor in separate PR?

Reply:
>Do you recommend we make these refactoring in this PR?

I did the refactoring in the same PR for the server auth. The changes were very similar but doing them uncovered one or two things that I had to accommodate elsewhere. If the changes aren't too big, I recommend doing them in this PR.

Reply:
Thanks @jpkrohling - finished updating the exporters kindly take a look and let me know.